common/firebase/database.ts

```diff
@@ -111,13 +120,37 @@ export interface CommonFirestore {
 
 type Collection = CollectionReference;
 
+const firestoreSettingsDataConverter: FirestoreDataConverter<Settings> = {
+  toFirestore(modelObject: Partial<Settings>) {
+    return modelObject;
+  },
+
+  fromFirestore(snapshot) {
+    const { canvasCalendar, completedOnboarding, theme } = snapshot.data() as Record<
+      string,
+      unknown
+    >;
+    if (
+      typeof canvasCalendar !== 'undefined' &&
+      canvasCalendar !== null &&
+      typeof canvasCalendar !== 'string'
+    ) {
+      throw new Error();
+    }
+    if (typeof completedOnboarding !== 'boolean') throw new Error();
+    if (theme !== 'light' && theme !== 'dark') throw new Error();
+    return { canvasCalendar, completedOnboarding, theme };
+  },
```

Comment:
we do some runtime type validation here to ensure that we never give the rest of the system incorrectly formatted object.